### DISCUSSION TITLE
DeepKeys

### DISCUSSION THREAD

**Discussion by nemo20000** [RISC-OS-Community/DME-Global-Shortcuts#2] [General]
As the author and maintainer of DeepKeys for over 23 years, I take a rather dim view of you **making use** of my ubiquitous module while intending to **subvert** its use longer term for some ideological concept of not-invented-here purity.

And to not even ask my permission!

You will also note that the licence for DeepKeys states:

  _DeepKeys can be distributed with any program, PD or commercial, providing that
 this documentation is included intact, and that the following credit is
 reproduced in any other documentation, manuals and/or help files:_

  _DeepKeys © Cerilica 1999-2020  Contact: <deepkeys@cerilica.co.uk>_

Are you planning on distributing your code _without_ DeepKeys while depending upon it, or will you be including the documentation as required by licence?

**Perhaps your time and effort is better served contributing things that don’t exist in the RISC OS world, rather than reimplementing things that have been used for two decades just because you don’t own the source?**

And if you are **content** to attempt to replace my work, presumably you will be content if I attempt to replace yours... just because I don’t own it? Yes? All fair?

**Comment by nemo20000** [comment]
DeepKeys is available here, with extensive documentation: http://deepkeys.20000.org

**Comment by pzaino** [comment]
Hi @nemo20000 ,
Thanks for your comment :)

I am sorry to hear the way you feel about the DME component in question. If you do not wish me to use your module, I won't, no worries.

The particular part that seems to have upset you, I quote it here:

> "As the author and maintainer of DeepKeys for over 23 years, I take a rather dim view of you making use of my ubiquitous module while intending to subvert its use longer term for some ideological concept of not-invented-here purity."

It's not about "purity" of any sort or way. It's mostly about:

What does it happens if you "end up under a bus" (which means any possible situation where the DeepKeys may not longer be available to this project) and we do not have the sources?

I am sure you can understand the situation, but, again, if this still upsets you then I won't use DeepKeys at all.

> And to not even ask my permission!

That is not correct, the document you are referring to is a design reference for something that is about to be coded and may or may not have the need to use DeepKeys hence, after investigating my options, I would have contacted you as I did with all the other authors for which their work is going to contribute to this entire project.

> You will also note that the licence for DeepKeys states:

Of course, in case of need to use DeepKeys as a must for this component, I would not only contact you to see if you'd be willing to allow me to use it, but obviously follow all the requirements of your license.

So I sense you may have jumped to conclusions here, hope this reply will clarify things.

All the best,
- Paolo

**Comment by nemo20000** [comment]
You have permission to include and distribute DeepKeys at no cost providing its documentation is included and copyright is acknowledged, just like everyone else. DeepKeys has been distributed with numerous third party programs for decades with no issue.

My objection is not to you using it, but you declaring that you are going to duplicate its functionality just so you can own the source code: “_we’ll replace this with a full Open Source substitute for DeepKeys, as we want to be able to maintain everything we release_”.

Inventing post facto a theoretical future unavailability is no defence for what is intellectual property theft. DeepKeys continues to be developed. It is not abandoned, unavailable, incompatible or undistributable.

I have already stated publicly in writing [here](https://twitter.com/nemo20000/status/1506283756814049288?s=20&t=iQAuZhLD8CR4cKiRjBrb3Q), [here](https://twitter.com/nemo20000/status/1506287689666072584?s=20&t=iQAuZhLD8CR4cKiRjBrb3Q) and [here](https://twitter.com/nemo20000/status/1506293069779509262?s=20&t=iQAuZhLD8CR4cKiRjBrb3Q) that you have permission to use and distribute DeepKeys. So your continued attempt to frame this as an objection is disingenuous.

You have permission to use and distribute it. But I will treat any attempt at Intellectual Property Theft extremely seriously, **starting** with a takedown notice to San Francisco 94107.

**Comment by pzaino** [comment]
I do not need to use DeepKeys. I'll write my own code to achieve the features I need, so you can sleep safe dreams and no one here is attempting to use or abuse of your work. Pace out man :)

All the best to you.
